platforms/android/run_module_app/build_apk.sh

```diff
@@ -0,0 +1,294 @@
+#!/bin/bash
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+#########################
+# Script input parameters
+#########################
+
+echo "=== A script for building iree-run-module apk ==="
+echo "This script wraps iree-run-module together with a"
+echo "specific IREE VM module and invocation information."
+echo ""
+echo "At a minimum, it expects the following env vars:"
+echo "* ANDROID_SDK_ROOT"
+echo "* ANDROID_NDK_ROOT"
+echo "* ANDROID_SDK_BUILD_TOOLS_VERSION"
+echo "See the script for details and more controls."
+echo "===-------------------------------------------==="
+echo ""
+
+print_usage_and_exit() {
+  echo "Usage: $0 <artifact-directory> "
+  echo "       --driver <driver>"
+  echo "       --module_file <input-module-file> "
+  echo "       --entry_function <entry-function> "
+  echo "       --inputs_file <input-buffer-file> "
+  exit 1
+}
+
+while (( "$#" )); do
```

Comment:
This loop scares me... can we use a real scripting language with good support for command line arguments instead of bash?
/me ducks for cover

Reply:
:D I wasn't able to find a good argparser for bash.. Don't really want to switch to Python just for the argparser given the majority of this script is about reading environment variables and calling external binaries and threading them together. Bash is good for such thing.